src/server_manager/install_scripts/install_server.sh

```diff
@@ -186,6 +188,7 @@ function get_random_port {
 
 function create_persisted_state_dir() {
   readonly STATE_DIR="$SHADOWBOX_DIR/persisted-state"
+  readonly STATE_HOST_PATH="${STATE_HOST_PATH:-$STATE_DIR}"
```

Comment:
I believe this isn't enough. Search for all uses of `STATE_DIR`. `generate_certificate`, for example, should change to write to the volume instead.

Maybe the certificate generation and other logic could migrate to the outline image and run on the container instead, so it has access to the volume.

Reply:
Oh, I guess that would work in your setup because the outer ubuntu container has the STATE_DIR pointing to the `outline_state` volume. But then you don't need the `STATE_HOST_PATH`.

But that docker-on-docker environment is not representative of real usage. Real usage would not have the nested containers, so the install script would have to have a way to access the created volume.